Sources/Commands/Options.swift

```diff
@@ -46,33 +46,43 @@ struct GlobalOptions: ParsableArguments {
 struct LocationOptions: ParsableArguments {
     init() {}
 
-    /// The custom build directory, if provided.
-    @Option(help: "Specify build/cache directory")
-    var buildPath: AbsolutePath?
+    /// The custom .build directory, if provided.
+    @Option(name: .customLong("scratch-space-path"), help: "Specify the scratch space directory (default .build)", completion: .directory)
+    var _scratchSpacePath: AbsolutePath?
 
-    @Option(help: "Specify the shared cache directory")
-    var cachePath: AbsolutePath?
+    @Option(name: .customLong("build-path"), help: .hidden)
+    var _deprecated_buildPath: AbsolutePath?
 
-    @Option(help: "Specify the shared configuration directory")
-    var configPath: AbsolutePath?
+    var scratchSpaceDirectory: AbsolutePath? {
+        self._scratchSpacePath ?? self._deprecated_buildPath
+    }
+
+    @Option(name: .customLong("cache-path"), help: "Specify the shared cache directory", completion: .directory)
+    var cacheDirectory: AbsolutePath?
 
-    @Option(help: "Specify the shared security directory")
-    var securityPath: AbsolutePath?
+    @Option(name: .customLong("config-path"), help: "Specify the shared configuration directory", completion: .directory)
+    var configurationDirectory: AbsolutePath?
 
-    /// The custom working directory that the tool should operate in (deprecated).
-    @Option(name: [.long, .customShort("C")])
-    var chdir: AbsolutePath?
+    @Option(name: .customLong("security-path"), help: "Specify the shared security directory", completion: .directory)
+    var securityDirectory: AbsolutePath?
 
     /// The custom working directory that the tool should operate in.
-    @Option(help: "Change working directory before any other operation")
-    var packagePath: AbsolutePath?
+    @Option(name: .customLong("package-path"), help: "Change working directory before any other operation", completion: .directory)
+    var _packageDirectory: AbsolutePath?
+
+    @Option(name: [.long, .customShort("C")], help: .hidden)
+    var _deprecated_chdir: AbsolutePath?
+
+    var packageDirectory: AbsolutePath? {
+        self._packageDirectory ?? self._deprecated_chdir
+    }
 
     /// The path to the file containing multiroot package data. This is currently Xcode's workspace file.
-    @Option(name: .customLong("multiroot-data-file"), completion: .file())
+    @Option(name: .customLong("multiroot-data-file"), help: .hidden, completion: .directory)
     var multirootPackageDataFile: AbsolutePath?
 
     /// Path to the compilation destination describing JSON file.
-    @Option(name: .customLong("destination"), completion: .file())
+    @Option(name: .customLong("destination"), help: .hidden, completion: .directory)
```

Comment:
What does this one even do?  Is it something that should still be supported in some fashion?

Reply:
Not saying it should, just raising the question since it will now get hidden.  That's probably a good thing though — there seems to have been organic growth among these options (which isn't a catastrophe) but it's nice to trim it down, as you're doing.

Reply:
it defines where the llbuild manifest get written to. there is not user facing feature I am aware of around this so my assumption is an accidental leak of a flag designed for for debugging

Reply:
Great, thanks for explaining.